trevismd/statannotations
# Use pyproject.toml with hatch

Modernize the building and testing framework:

* use pyproject.toml
* use hatch environments
* ~~add test_script to test plotting figures~~ -> added in #155 
* untrack `docs/build`

To build: `hatch build` or `python -m build`
To bump version: `hatch version minor`
To test: `hatch run tests:run`
To test with coverage: `hatch run tests:run-cov && hatch run coverage:run`
To build and test docs: `hatch run docs:all`
To run examples: `hatch run examples:run -- --plot_type=boxplot --data_type=df_with_group_and_hue`


To test #155 I had to write the `test_script.py` file and it was very convenient to use `hatch` environments for testing. So I thought I should actually make a PR about it :)
